### PR TITLE
Api v0

### PR DESCRIPTION
**Pull request by Terrtia**
Add users managements + API (sensor registration)

Update ``v0.3`` to ``v0.4``:
- ``. ./D4ENV/bin/activate``
- ``pip3 install bcrypt``
- ``pip3 install flask-login``
- ``./LAUNCH -lrv``
- ``cd web``
- ``./create_default_user.py``

### CODE REVIEW

File: server/web/blueprints/restApi.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+def verify_token(token):
+    if len(token) != 41:
+        return False
+
+    if not check_token_format(token):
+        return False
+
+    if r_serv_db.hexists('user:tokens', token):
+        return True
+    else:
```
Comment (adulau) [inline server/web/blueprints/restApi.py:69]: You can remove the else and return False by default to avoid edge cases.

Another (potential) issue is the side-channel attack to test token, maybe adding a small random delay could help.

File: server/web/blueprints/settings.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+                decode_responses=True)
+
+# ============ FUNCTIONS ============
+
+def one():
+    return 1
+
+def check_email(email):
+    result = email_regex.match(email)
+    if result:
```
Comment (adulau) [inline server/web/blueprints/settings.py]: Maybe simplify the code to return directly the result?

File: server/web/templates/server_management.html [hunk 3]
```diff
@@ -19,22 +19,7 @@
...
-	    </li>
-	      <li class="nav-item" mr-3>
-	        <a class="nav-link mr-3" href="{{ url_for('sensors_status') }}">Sensors Status</a>
-	      </li>
-	      <li class="nav-item mr-3">
-	        <a class="nav-link active" href="{{ url_for('server_management') }}" tabindex="-1" aria-disabled="true">Server Management</a>
-	      </li>
-	    </ul>
-	</nav>
+	{% include 'navbar.html' %}
```
Comment (adulau) [inline server/web/templates/server_management.html:22]: :+1: I love the navbar simplification!

**Review by adulau — CHANGES_REQUESTED** [review]
Looks very good. Just some minor stuff. Thanks a lot for the work!

**Comment by Terrtia** [comment]
API + login protection added:
- brute force
- side-channel attack
